spec/serializers/provider_serializer_spec.rb

```diff
@@ -84,4 +84,78 @@
 
     it { should eq provider.ucas_preferences.send_application_alerts_before_type_cast }
   end
+
+  describe 'contacts' do
+    describe 'admin contact' do
+      let(:provider) do
+        create :provider,
+               contacts: [admin]
+      end
+      let(:admin) { create(:contact, type: 'admin') }
+
+      subject { serialize(provider)['contacts'].find { |c| c[:type] == 'admin' } }
+
+      its([:name]) { should eq admin.name }
+      its([:email]) { should eq admin.email }
+      its([:telephone]) { should eq admin.telephone }
+    end
+
+    describe 'utt contact' do
```

Comment:
context vs describe

Reply:
imo

* **describe**: this is what we're testing (kind of like a noun)
* **context**: this is what state the system is in (kind of like an adjective)

Reply:
Then I'm just getting a feeling there is a lack of intent or the understanding of significant of what it is `testing` or its halfway point, from line 88 to 160 (Nevermind the dupes).

Therefore its more aligned with `context` then `describe`.

Reply:
I think if I put all your comments together, mix them up, and then reduce them on the hob for a bit, stirring constantly, I come up with something like:

> [You don't need to test every type of contact here, since the code behaves the same for each type and doesn't differentiate, so why bother?]

So we should be able to test for a single contact type, which makes sense to me. 😸

Reply:
👍

Reply:
Thanks for all the feedback. Will do.